### PR TITLE
Adapt to work in workload clusters

### PR DESCRIPTION
**Pull request by pipo02mix**
## Checklist

- [x] Update CHANGELOG.md

### CODE REVIEW

File: helm/dex-app/templates/_helpers.tpl [hunk 1]
```diff
@@ -69,3 +69,11 @@ Selector dex-k8s-authenticator giantswarm labels
 app.kubernetes.io/name: dex-k8s-authenticator-giantswarm
 app.kubernetes.io/component: dex-k8s-authenticator-giantswarm
 {{- end -}}
+
+{{- define "giantswarm-connector" -}}
+{{- if .Values.oidc.giantswarm.connectorConfig.clientID -}}
```
Comment (kopiczko) [inline helm/dex-app/templates/_helpers.tpl:74]: Why wouldn't that be defined?

Comment (pipo02mix) [reply helm/dex-app/templates/_helpers.tpl:74]: because my intention here is to open to all type of clusters. So I have defaulted GS connector

Comment (pipo02mix) [reply helm/dex-app/templates/_helpers.tpl:74]: default values.yaml were already 
```
  giantswarm:
    connectorConfig:
      clientID: ""
      clientSecret: ""
      team: ""
    write_all_group: ""```

Comment (kopiczko) [reply helm/dex-app/templates/_helpers.tpl:74]: TBH it isn't great to have `giantswarm` in values.yaml

Comment (pipo02mix) [reply helm/dex-app/templates/_helpers.tpl:74]: but this is the same chart we use for MC, so we don't need to maintain two charts. I don't want to touch any values so I ensure it does not impact MC deployments

File: .circleci/config.yml [hunk 2]
```diff
@@ -79,3 +79,14 @@ workflows:
...
+      # make it available to customers (only tags)
+      - architect/push-to-app-catalog:
+          name: "package and push dex-app chart"
+          app_catalog: "giantswarm-playground-catalog"
+          app_catalog_test: "giantswarm-playground-test-catalog"
+          chart: "dex-app"
+          # Trigger job on git tag.
+          filters:
+            tags:
+              only: /^v.*/
```
Comment (marians) [inline .circleci/config.yml:92]: I understand that this adds it to the Playground catalog. Should we have the app in two catalogs in the long term? If no, I would create an issue for team rainbow to remove it from the CP catalog.

File: helm/dex-app/templates/_helpers.tpl (hunk 1 above)
Comment (marians) [reply helm/dex-app/templates/_helpers.tpl:74]: If Pawel's concern is mainly usability, I recommend to add some commenting that explains why this giantswarm entry is here, and what it's for.

File: .circleci/config.yml (hunk 2 above)
Comment (pipo02mix) [reply .circleci/config.yml:92]: It is a good question. I am not 100% sure how this work. The MCs run dex via shared collection, which I am not sure if always pull from the Control Plane catalog. Something to check later. Do you want me to create the issue?

File: helm/dex-app/templates/_helpers.tpl (hunk 1 above)
Comment (pipo02mix) [reply helm/dex-app/templates/_helpers.tpl:74]: ok I add it now. I get the idea but I don't want to touch now the default values format, it is risky.

-

File: .circleci/config.yml (hunk 2 above)
Comment (marians) [reply .circleci/config.yml:92]: No, I can create it. Thanks.

Comment (marians) [reply .circleci/config.yml:92]: Created: https://github.com/giantswarm/giantswarm/issues/19065

Also it just occured to me that as a Playground app, dex-app should be added to the repositories list https://github.com/giantswarm/docs/blob/master/scripts/aggregate-changelogs/config.yaml#L9 to appear in our aggregated changelogs.

Comment (pipo02mix) [reply .circleci/config.yml:92]: https://github.com/giantswarm/docs/pull/1138/files

Comment (marians) [reply .circleci/config.yml:92]: Nice, thanks

**Comment by pipo02mix** [comment]
@kopiczko @marians a customer is waiting for it, can you approve or give me more feedback?

**Review by marians — APPROVED** [review]
Unblocking

**Comment by pipo02mix** [comment]
I have checked the manifest produced with some MC values to verify does not break existing dex deployments. Going ahead

**Pull request merged**